iodata/formats/pdb.py

```diff
@@ -0,0 +1,132 @@
+# IODATA is an input and output module for quantum chemistry.
+# Copyright (C) 2011-2019 The IODATA Development Team
+#
+# This file is part of IODATA.
+#
+# IODATA is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 3
+# of the License, or (at your option) any later version.
+#
+# IODATA is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, see <http://www.gnu.org/licenses/>
+# --
+"""PDB file format.
+
+There are different formats of pdb files.
+"""
+
+
+from typing import TextIO, Iterator
+
+import numpy as np
+
+from ..docstrings import (document_load_one, document_load_many, document_dump_one,
+                          document_dump_many)
+from ..iodata import IOData
+from ..periodic import sym2num, num2sym
+from ..utils import angstrom, LineIterator
+
+
+__all__ = []
+
+
+PATTERNS = ['*.pdb']
+
+
+@document_load_one("PDB", ['atcoords', 'atnums'], ['title'])
+def load_one(lit: LineIterator) -> dict:
+    """Do not edit this docstring. It will be overwritten."""
+    nums = []
+    coords = []
+    attypes = []
+    restypes = []
+    molecule_found = False
+    while True:
+        try:
+            line = next(lit)
+        except StopIteration:
+            break
+        if line.startswith("TITLE") or line.startswith("COMPND"):
+            words = line.split()
+            title = words[1:]
+        else:
+            title = "Pdb file from Iodata"
+        if line[0:4] == "ATOM" or line[0:6] == "HETATM":
+            try:
+                # Try reading element from position 77:79 referenced in pdb format
+                words = line[77:79].split()
+                if not words:
+                    # If not guess it from the atom type
+                    words = line[13:16].split()
+                symbol = words[0].title()
+                nums.append(sym2num.get(symbol, sym2num.get(symbol[0], None)))
+            except ValueError:
+                # When the position 77:79 does not exist also use the atom type
+                symbol = line[13:16].title()
+                nums.append(sym2num.get(symbol, sym2num.get(symbol[0], None)))
+            x = float(line[30:38])
+            y = float(line[38:46])
+            z = float(line[46:54])
+            coords.append([x, y, z])
+            attypes.append(line[13:16].strip())
+            restypes.append(line[18:20].strip())
+            molecule_found = True
+        if line[0:3] == "END" and molecule_found:
+            atnums = np.array(nums)
+            atcoords = np.array(coords) * angstrom
+            attypes = tuple(attypes)
+            restypes = tuple(restypes)
```

Comment:
Similar suggestion:
```suggestion
            restypes = np.array(restypes, dtype="U3")
```

Reply:
The reason I used a tuple is that I thought it expects it to be a tuple when you assign it to the atffparams dictionary.

Reply:
The `IOData` class will not enforce a specific type for the values of the `atffparams` dictionary. Any type can be used in principle, but the documentation says the values of this dictionary are arrays: https://github.com/theochem/iodata/blob/58d5e4bd1fb83409ca1e6ceeb474c4358c6ddc10/iodata/iodata.py#L80-L85

I can't think of a use case where tuples would have an advantage over arrays, but if there is a technical reason, please let me know. Arrays would in practice allow for faster operations, e.g. displacing all atoms of a given type would be relatively fast and easy with arrays: 

```python
mol.atcoords[mol.atffparams["attypes"] == "CA"] += np.array([1.0, 0.0, 0.0])
```

Reply:
You are right. I had previously tried with a list if I remember correctly and that gave an error. I changed both restypes and attypes to arrays.
Thanks